dashboard/lib/clients/lti_advantage_client.rb

```diff
@@ -0,0 +1,22 @@
+class LtiAdvantageClient
+  include LtiAccessToken
+
+  def initialize(client_id, issuer)
+    raise ArgumentError unless client_id && issuer
+    @client_id = client_id
+    @issuer = issuer
+  end
+
+  # Call the LTI Advantage API to get the membership (roster) for a given context (course/section).
+  # The URL for the API will vary between LMS platforms, and is initially contained in the launch context.
+  # See the LTI spec for details:
+  # https://www.imsglobal.org/spec/lti-nrps/v2p0/
+  def get_context_membership(url)
+    headers = {
+      'Accept' => Policies::Lti::MEMBERSHIP_CONTAINER_CONTENT_TYPE,
+    }
+    res = HTTParty.get(url, headers: headers.merge('Authorization' => "Bearer #{get_access_token(@client_id, @issuer)}"))
```

Comment:
What happens if `get_access_token` raises an exception here, will it get swallowed? Should we try to rescue the exception, or are we ok with letting the exception on line 19 tell us there may be something wrong with the access token?

Reply:
The exception will actually bubble up - I tried this with an empty token and I got it threw the proper exception. My gut was not to rescue since we want it to fail if we failed to get the access token, but I also don't know if there's a Ruby way to do it. In JS I'd probably let this function throw an error then catch and handle it in the code where I'm calling it.

Reply:
Oh cool, yeah I wasn't sure how Ruby handled this, if it's bubbling up and can be handled in the code that calls this method, thats great 👍